# To what extent does allele- and/or isoform-specific expression occur in _S. marinoi_?

Are any of these condition-specific, particularly with regard to the resting stages?

## Previous papers on ASE/ISE in diatoms

### ASE

|                                Study                                  |        Species        |                                 Results                                  |                           Notes                           |
|-----------------------------------------------------------------------|-----------------------|--------------------------------------------------------------------------|-----------------------------------------------------------|
| (Mock _et al._, 2017)[https://doi.org/10.1038/nature20803]            | _Fragilariopsis cylindrus_ | "approximately 66% (4,030) of diverged alleles showed greater than fourfold significant differential expression relative to optimal nutrient-replete growth and approximately 45% (2,730) of divergent alleles showed greater than fourfold unequal bi-allelic expression between allele 1 and allele 2 in at least one RNA-seq experiment" ||
| (Nomaguchi _et al._, 2018)[https://doi.org/10.1186/s12864-018-4691-0] | _Fistulifera solaris_ | "subgenomic preferences in approximately 61% of homoeologous gene pairs" | Allopolyploid so, may be a different case in _S. marinoi_ |
| (Hoguin _et al._, 2021)[https://doi.org/10.1038/s41598-021-82529-1]   | _Phaeodactylum tricornutum_ | "1395 (~ 11%) genes were categorized as ‘BAE’, 2662 (~ 22%) genes were characterized as ‘ASE’, and 129 (~ 1%) genes were denoted ‘MAE’." ||

* BAE = biallelic expression (low % expression bias)
* ASE = allele-specific expression (moderate % expression bias)
* MAE = monoallelic expression (high % expression bias)

### ISE

|                                Study                                  |        Species        |                                 Results                                  |                           Notes                           |
|-----------------------------------------------------------------------|-----------------------|--------------------------------------------------------------------------|-----------------------------------------------------------|
| (Rastogi _et al._, 2018)[https://doi.org/10.1038/s41598-018-23106-x]  | _Phaeodactylum tricornutum_ | (See below) ||
| (Goldman _et al._, 2019)[https://doi.org/10.1371/journal.pone.0222325] | _Thalassiosira pseudonana_ | "we identified a non-canonical splicing of transcripts encoding triose-phosphate isomerase [...] generating transcript isoforms that would encode proteins with and without an active site." | Focused on a select group of genes |
| (Cheng _et al._, 2021)[https://doi.org/10.3390/md19070392]      | _Thalassiosira weissflogii_ | "3113 AS events were detected, with the genes containing two isoforms ranked the highest (1105)" | Used PacBio; no ref. genome       |

Points of interest from Rastogi _et al._ paper:
* "From the 12,177 Phatr3 gene models, 2,924 (~24%) and 2,444 (~20%) genes undergo IR and ES, respectively."
* "A total of 1,335 (~11%) genes are found to undergo both ES and IR, hence can perform alternative splicing."
* "Surprisingly, we found that genes that can undergo intron-retention are more highly expressed than genes that do not show alternative splicing."
* "To further assess the biological role of alternative splicing, we identified 1341 genes showing IR and 1099 genes with ES during an 18-hour time course under nitrogen-free growth conditions"
  * "we found a significant increase in the expression levels of genes undergoing intron retention"
  * "we observed a significant decrease in the expression of genes undergoing ES across different time-points of nitrogen-free growth"
* "We identified 81 GO categories which were significantly over-represented in either IR or ES across all or dif- ferent time-points and with plausible functions in tolerating nitrogen starvation"
